### PR TITLE
bug: Decimals crash bug

### PR DESCRIPTION
**Pull request by Tulun**
**Motivation**
Certain string values caused app to crash

**Summary**

Capture the error and show an invalid string error when the user adds too many decimals or shows a single decimal with no number.

Screen Shot 2021-08-13 at 10 26 23 AM

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: components/lsp/MintForm.tsx [hunk 1]
```diff
@@ -77,13 +79,18 @@ const MintForm: FC<Props> = ({
...
+      if (
+        amount &&
+        toWeiSafe(amount, Number(collateralDecimals)).gt(collateralBalance)
+      ) {
+        setShowMintError(INSUFFICIENT_COLLATERAL_ERROR);
+      } else if (showMintError) {
+        setShowMintError("");
+      }
+    } catch (err) {
+      console.log("err", err);
```
Comment (daywiss) [inline components/lsp/MintForm.tsx:92]: can prop remove this

File: components/text-input/TextInput.tsx [hunk 2]
```diff
@@ -33,9 +37,13 @@ const TextInput: FC<Props> = ({
           type="text"
           value={value}
           onChange={(e) => {
+            if (customOnChange) {
```
Comment (daywiss) [inline components/text-input/TextInput.tsx:40]: i dont think we need the chnages to this file in this pr

File: components/lsp/MintForm.tsx (hunk 1 above)
Comment (Tulun) [reply components/lsp/MintForm.tsx]: Removed.

File: components/text-input/TextInput.tsx (hunk 2 above)
Comment (Tulun) [reply components/text-input/TextInput.tsx:40]: I think it's ok.

It's a minor change that doesn't affect functionality. I can undo it if you really want, but it should allow an expanded use case that could easily be useful later.

**Review by daywiss — APPROVED** [review]

File: components/lsp/helpers.ts [hunk 3]
```diff
@@ -39,3 +39,15 @@ export const onlyAllowNumbersAndDecimals = (
     return false;
   }
 };
+
+export const INSUFFICIENT_COLLATERAL_ERROR =
+  "You don't have enough collateral to mint.";
+
+export const INVALID_STRING_ERROR =
+  "String is not convertable to a wei number.";
```
Comment (mrice32) [inline components/lsp/helpers.ts]:
```suggestion
  "String is not convertable to a number.";
```

**Review by mrice32 — APPROVED** [review]
Only comment is that wei number is probably a little confusing. I would suggest just "number".

**Comment by Tulun** [comment]
> Only comment is that wei number is probably a little confusing. I would suggest just "number".

Fair enough, updated.

**Pull request merged**